proto/streamlit/proto/Arrow.proto

```diff
@@ -17,14 +17,33 @@
 syntax = "proto3";
 
 message Arrow {
-  bytes data = 1;
-  Styler styler = 2;
-  // width in CSS points
-  uint32 width = 3;
-  // height in CSS points
-  uint32 height = 4;
+  // The id of the widget, this is required if the dataframe is editable
+  string id = 1;
+  // The serialized arrow dataframe
+  bytes data = 2;
+  // Pandas styler information
+  Styler styler = 3;
+  // Width in CSS points
+  uint32 width = 4;
+  // Height in CSS points
+  uint32 height = 5;
   // If True, will overwrite the dataframe width to fit to container.
-  bool use_container_width = 5;
+  bool use_container_width = 6;
+  // Colum configuration as JSON
+  string columns = 7;
+  // Activate table editing
+  EditingMode editing_mode = 8;
+  // Deactivates editing
+  bool disabled = 9;
+  // The form ID of the widget, this is required if the dataframe is editable
+  string form_id = 10;
+
+  // Available editing modes:
+  enum EditingMode {
+    READ_ONLY = 0; // Read-only table.
+    FIXED = 1; // Activates editing but only allow editing of existing cells.
+    DYNAMIC = 2; // Activates editing and allow adding & deleting rows.
+  }
```

Comment:
Per our Slack discussion, let's change this to:

- Not change message IDs for existing data
- Not reuse old message IDs for adding new data.

(All existing data should keep the same ID, and new data should just get new, never-been-used-before IDs)

Reply:
Done ✅